### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by rasendubi**
Includes fixes for:
- doxygen comments
- macro definitions with do { ... } while (0)
- difference between () and (void) functions
- code formatting
- code duplication

### CODE REVIEW

**Comment by forGGe** [comment]
Looks good for me, but someone else has to approve.

**Comment by ohamov** [comment]
LGTM

File: client/client-multi/client-c/src/kaa/collections/kaa_list.c [hunk 1]
```diff
@@ -211,29 +211,15 @@ kaa_error_t kaa_list_remove_first(kaa_list_t *list, match_predicate pred, void *
...
+    kaa_list_remove_at(list, it, deallocator);
+    return it ? KAA_ERR_NONE : KAA_ERR_NOT_FOUND;
 }
 
 void kaa_list_set_data_at(kaa_list_node_t *it, void *data, deallocate_list_data deallocator)
 {
     KAA_RETURN_IF_NIL(it, );
     if (deallocator) {
-        (*deallocator)(it->data);
+        deallocator(it->data);
```
Comment (akokhanovskyi) [inline client/client-multi/client-c/src/kaa/collections/kaa_list.c:222]: Is there a unit test for this function?

**Comment by akokhanovskyi** [comment]
Generally, looks safe and useful - however I would highly appreciate a unit test for kaa_list_set_data_at()

File: client/client-multi/client-c/src/kaa/collections/kaa_list.c (hunk 1 above)
Comment (rasendubi) [inline client/client-multi/client-c/src/kaa/collections/kaa_list.c:222]: I'm not a fan of testing the compiler. (*deallocator)(x) is guaranteed to be the same thing as to deallocator(x).

**Comment by ghost** [comment]
LGTM

File: client/client-multi/client-c/src/kaa/collections/kaa_list.c (hunk 1 above)
Comment (akokhanovskyi) [inline client/client-multi/client-c/src/kaa/collections/kaa_list.c:222]: Okay. Unit tests are, nonetheless, greatly appreciated.

**Comment by ghost** [comment]
Please, resolve conflicts.

**Pull request merged**